### DISCUSSION TITLE
Picking overlapping bitmap layers

### DISCUSSION THREAD

**Discussion by alexandruandrei91** [visgl/deck.gl#5574] [Q&A]
Hi,

We have a couple of overlapping layers that extend the bitmap layer ([Demo](https://equinor.github.io/webviz-subsurface-components/), [Code](https://github.com/equinor/webviz-subsurface-components/tree/master/src/lib/components/DeckGLMap)) and I need to get the value under the cursor for all of them.
I tried using DeckGL's onHover, but I only get the topmost layer. I've also tried using the react events and call `pickMultipleObjects`, but I get the following error when using it with the Bitmap layer:
```
Uncaught TypeError: colors is undefined
    _disablePickingIndex layer.js:673
    disablePickingIndex layer.js:664
    _pickClosestObject deck-picker.js:252
    pickObject deck-picker.js:64
    _pick deck.js:364
    pickMultipleObjects deck.js:321
    pickMultipleObjects deckgl.js:89
    onClick DeckGLMap.jsx:68
```

Other thing I've tried was adding the `instancePickingColors` attribute back to the bitmap layer, but with no luck.
I've also tried overriding `disablePickingIndex` to change a uniform in the fragment shader and cause it to return vec4(0.0) in the picking drawing pass, and change it back in `restorePickingColors`, no success with that either. I didn't get the error anymore, but I've only got the topmost layer.

Any suggestions on how can I pick multiple bitmap layers?

**Comment by Pessimistress** [comment]
Thanks for reporting this. Would you open a bug? 

@ibgreen this layer no longer has an invalid picking color so it cannot be removed in multi picking pass.

**Reply by alexandruandrei91** [reply]
Sure, I'll do that. Should I mark this as "answered" and move the discussion on the issue?

edit: The issue is here: https://github.com/visgl/deck.gl/issues/5576
